core/interfaces/src/main/java/info/nightscout/interfaces/ApsMode.kt

```diff
@@ -0,0 +1,19 @@
+package info.nightscout.interfaces
+
+enum class ApsMode() {
+    OPEN,
+    CLOSED,
+    LGS,
+    UNDEFINED;
+
+    companion object {
+
+        fun secureValueOf(stringValue: String): ApsMode {
```

Comment:
i'd prefer something like
`fun fromString(stringValue: String?) = values().firstOrNull { it.text == stringValue}  ?: UNDEFINED`

Reply:
Done!